ios/Exponent/Versioned/Core/Api/EXScreenOrientation.m

```diff
@@ -33,37 +41,210 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId
+                                  @"PORTRAIT_UP" : @(PORTRAIT_UP_LOCK),
+                                  @"PORTRAIT_DOWN" : @(PORTRAIT_DOWN_LOCK),
+                                  @"LANDSCAPE" : @(LANDSCAPE_LOCK),
+                                  @"LANDSCAPE_LEFT" : @(LANDSCAPE_LEFT_LOCK),
+                                  @"LANDSCAPE_RIGHT" : @(LANDSCAPE_RIGHT_LOCK),
+                                  @"OTHER" : @(OTHER_LOCK),
+                                  @"ALL_BUT_UPSIDE_DOWN": @(ALL_BUT_UPSIDE_DOWN_LOCK)
+                                  };
+  });
+  return strToOrientationLockDict;
+}
+
++ (NSDictionary *)getOrientationLockToStrDict
+{
+  static NSMutableDictionary* orientationLockToStrDict = nil;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    orientationLockToStrDict = [[NSMutableDictionary alloc] init];
+    NSDictionary * strToOrientationLock = [EXScreenOrientation getStrToOrientationLockDict];
+    for(NSString * str in strToOrientationLock) {
+      NSNumber * wrappedOrientationLock = [strToOrientationLock objectForKey:str];
+      orientationLockToStrDict[wrappedOrientationLock] = str;
+    }
+  });
+  return orientationLockToStrDict;
+}
+
+// Will be called when this module's first listener is added.
+-(void)startObserving {
+  hasListeners = YES;
+  [_kernelOrientationServiceDelegate addOrientationChangeListener:self.experienceId subscriberModule:self];
+}
+
+// Will be called when this module's last listener is removed, or on dealloc.
+-(void)stopObserving {
+  hasListeners = NO;
+  [_kernelOrientationServiceDelegate removeOrientationChangeListener:self.experienceId];
+}
+
+- (void) handleScreenOrientationChange: (UITraitCollection *)traitCollection {
+  
+  UIInterfaceOrientationMask orientationMask = [_kernelOrientationServiceDelegate supportedInterfaceOrientationsForVisibleApp];
+  if (hasListeners) {
+    EXOrientationLock orientationLock = [self orientationLockNativeToJS:orientationMask];
+    [self sendEventWithName:@"expoDidUpdateDimensions" body:@{
+                                                              @"orientationInfo": [self getOrientationInformation:traitCollection],
+                                                              @"orientationLock": [self orientationLockToString:orientationLock]
+                                                              }];
+  }
+}
+
+- (NSDictionary *) getOrientationInformation: (UITraitCollection *)traitCollection {
+  EXOrientation orientation = [self traitCollectionToOrientation:traitCollection];
+  return @{
+           @"orientation": [self orientationToString:orientation],
+            @"verticalSizeClass": [self sizeClassToString: traitCollection.verticalSizeClass],
+            @"horizontalSizeClass": [self sizeClassToString: traitCollection.horizontalSizeClass]
+            };
+}
+
+- (NSArray<NSString *> *)supportedEvents {
+  return @[@"expoDidUpdateDimensions", @"expoDidUpdatePhysicalDimensions"];
```

Comment:
is `expoDidUpdatePhysicalDimensions` still used?

Reply:
nope, good catch!